Add additional search view pages to the Nuxt app

obulat opened the pull request:
~~This PR is based on #3136, and will be undrafted after that PR is merged.~~
<!-- prettier-ignore -->
## Fixes
<!-- If PR doesn't fully resolve the issue, replace 'Fixes' below with 'Related to'. -->
<!-- If there is no issue being resolved, please consider opening one before creating this pull request. -->

Fixes #2858 by @obulat
Fixes #2774 by @obulat 

## Description
<!-- Concisely describe what the pull request does. -->
<!-- Add screenshots, videos, or other media to show the problem and the solution when appropriate. -->

With this PR, you can view the tag/source/creator collections in the Nuxt app.

To show the collections, we add `_tag.vue` and `_source.vue` pages to both `image` and `audio`, with dynamic routes. These pages check the path to make sure that the path has the necessary parameters: tag name for the tag page; an existing source for source/creator pages; `/creator/` path part and the creator name for the creator pages. 

In the first iteration, I used a single page with a fully-dynamic path (`/_mediaType/_collection/_.vue`) that would then validate the `mediaType`, `collection` and the collection parameters. However, because `_mediaType` can accept any string path, it would also match the locale routes such as `/ar/non_existent_page`), so I split this single page into 4.

The collection pages have the collection headers (using `VCollectionHeader` from #3365) and either a `VAudioCollection` or `VImageGrid`, as well as the "Load more" button. When you navigate from the collection page to a single result page, you also get the "Back to results" link at the top that goes back to the collection page.

This PR adds a `strategy` and `collectionParams` props to the `search` store. Whenever we set the `searchTerm` (which is only available for search, not collection), we reset the `strategy` to `default` and nullify the `collectionParams`.

The API service had to be updated to accept a `slug` for paths like `/tag/tag_name`, `/source/source_name`, `/source/source_name/creator/creator_name`, which are added to the `/<mediaType>` path.
The media store was updated to use the necessary slug from the search store `collectionParams`.

## Testing Instructions
<!-- Give steps for the reviewer to verify that this PR fixes the problem; or delete this section entirely. -->
Run the app using `just frontend/run dev`. Go to `/preferences` page and turn on the `additional_search_views` feature flag.

The easiest way to land on a collection page is from the single result page. Click on the source, creator or tag buttons - and you will be taken to the corresponding page.
Check that in non-English locales, the links have the correct locale path, and when navigating, the locale is kept. 
Check that if you go from a collection page to a single result page, you can go back to the collection page using "Back to results" page.
Check that if you go back to homepage from the collection page, you can search for something.

## Checklist
<!-- Replace  the [ ] with [x] to check the boxes. -->

- [ ] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [ ] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [ ] My commit messages follow [best practices][best_practices].
- [ ] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [ ] I tried running the project locally and verified that there are no visible errors.
- [ ] I ran the DAG documentation generator (if applicable).

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin
<!-- You must read and understand the following attestation. -->

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

obulat commented on frontend/src/components/VLoadMore.vue:
```diff
@@ -78,6 +86,14 @@ export default defineComponent({
         resultPage: currentPage.value || 1,
       })
 
+      router.push({
+        path: route.value.path,
+        query: {
+          ...route.value.query,
+          page: `${currentPage.value + 1}`,
+        },
+      })
+
```
```suggestion
```

obulat commented on frontend/src/components/VLoadMore.vue:
```diff
@@ -18,7 +18,7 @@ import { computed, defineComponent, onMounted, ref, watch } from "vue"
 import { useElementVisibility } from "@vueuse/core"
 
-import { useRoute } from "@nuxtjs/composition-api"
+import { useRoute, useRouter } from "@nuxtjs/composition-api"
```
```suggestion
import { useRoute } from "@nuxtjs/composition-api"
```

fcoveram reviewed (changes requested):
It works great 👏 

I just noticed one area of improvement:
* When `VCollectionHeader` has a padding of 40px on the sides (`xs / sm / md`), top padding looks better with 8px.
* When `VCollectionHeader` has a padding of 24px on the sides (`lg / xl / 2xl`), top padding looks better with 0px.

On the other hand and something out of this scope, I noticed that loading these views keeps the page empty for a few milliseconds. It seems good idea showing the [media's skeleton](https://docs.openverse.org/storybook/?path=/story/components-skeleton--all-tab). Here is a video recording of what I see.

https://github.com/WordPress/openverse/assets/895819/bbb35dc6-2a85-4f28-aa0f-275d9cb8b40b

If you think is a good idea to include. I can request it in a new ticket.

dhruvkb reviewed (approved):
The pages work well, when they do work. I have nothing to add to @fcoveram's review, LGTM from my side!

AetherUnbound reviewed (commented):
THIS IS SO COOOOOL! 🤩 It's amazing to see this come together on the frontend!! I have a few questions, but also - should we have any snapshots as part of this PR too?

AetherUnbound reviewed (approved):
I don't actually have anything additional that's blocking 😄

obulat commented:
> On the other hand and something out of this scope, I noticed that loading these views keeps the page empty for a few milliseconds. It seems good idea showing the [media's skeleton](https://docs.openverse.org/storybook/?path=/story/components-skeleton--all-tab). Here is a video recording of what I see.

I updated the paddings, @fcoveram . As for the loading skeletons: they were broken on `main` branch because a variable was renamed. I fixed the error on `main` and rebased this branch, so now you can see the loading skeletons when the media is loading.

AetherUnbound commented on frontend/src/utils/validate-collection-params.ts:
```diff
@@ -0,0 +1,87 @@
+  params,
+  $pinia,
+}: {
+  firstParam: "tag" | "source"
```
Got it, hmm. I'm not sure I have a better term 😅 I think I also missed while looking at this that `creator` is the second part.

fcoveram reviewed (changes requested):
I noticed one odd spacing in `VCollectionHeader` that can look better with the following two changes:
* Audio results of source, tag, and author in all breakpoints, margin-bottom looks better with `8px`. I refer to [this area](https://share.cleanshot.com/ZC2FXxlm).
* Image results of source, tag, and author in all breakpoints, margin-bottom looks better with `16px`. I refer to [this area](https://share.cleanshot.com/ZKY9kd3D).

Pardon the picky territory my suggestions belong to, but QA is the perfect stage for adjustments.

obulat commented:
> I noticed one odd spacing in `VCollectionHeader` that can look better with the following two changes:
> 
> * Audio results of source, tag, and author in all breakpoints, margin-bottom looks better with `8px`. I refer to [this area](https://share.cleanshot.com/ZC2FXxlm).
> * Image results of source, tag, and author in all breakpoints, margin-bottom looks better with `16px`. I refer to [this area](https://share.cleanshot.com/ZKY9kd3D).
> 
> Pardon the picky territory my suggestions belong to, but QA is the perfect stage for adjustments.

Thank you for re-reviewing! I've updated the paddings.

I also noticed that the collection pages really didn't render on the server at all, not just because of some local problems. Thank you for flagging that, @fcoveram. I fixed the logic of the collection page parameter validation, and now they do render on the server.
I've added the e2e tests, but didn't add visual regression tests to not make the tests run too long.

fcoveram reviewed (approved):
It looks great ✨🚀
